ride-share.rb

```diff
@@ -0,0 +1,235 @@
+        cost: 15,
+        rate: 1,
+      },
+      {
+        date: "5th Feb 2016",
+        rider_id: "RD0066",
+        cost: 35,
+        rate: 3,
+
+      },
+    ],
+  },
+  {
+    id: "DR0003",
+    rides: [
+      {
+        date: "4th Feb 2016",
+        rider_id: "RD0066",
+        cost: 5,
+        rate: 5,
+      },
+      {
+        date: "5th Feb 2016",
+        rider_id: "RD0003",
+        cost: 50,
+        rate: 2,
+      },
+    ],
+  },
+  {
+    id: "DR0004",
+    rides: [
+      {
+        date: "3rd Feb 2016",
+        rider_id: "RD0022",
+        cost: 5,
+        rate: 5,
+      },
+      {
+        date: "4th Feb 2016",
+        rider_id: "RD0022",
+        cost: 10,
+        rate: 4,
+      },
+      {
+        date: "5th Feb 2016",
+        rider_id: "RD0073",
+        cost: 20,
+        rate: 5,
+      },
+    ],
+  },
+]
+
+# 1 The number of rides each driver has given
+def total_rides(drivers)
+  drivers.each do |driver|
+    rides = driver[:rides].length
+    puts "Driver ID:#{driver[:id]} has given #{rides} rides."
+  end
+end
+
+# 2 The total amount of money each driver has made
+def total_money_made(drivers)
+  drivers.each do |driver|
+    total_cost = 0
+    driver[:rides].each do |ride_info|
+      total_cost += ride_info[:cost]
+    end
+    puts "The total amount of money driver ID:#{driver[:id]} has made is $#{total_cost}."
+  end
+end
+
+# 3 The average rating for each driver
+def average_rating(drivers)
+  drivers.each do |driver|
+    total_rating = 0
+    driver[:rides].each do |ride_info|
+      total_rating += ride_info[:rate]
+    end
+    average = (total_rating / driver[:rides].length)
+    puts "The average rating for driver ID: #{driver[:id]} is #{average}."
+  end
+end
+
+# 4 Which driver made the most money?
+def highest_money(drivers)
+  driver_profits = drivers.map do |driver|
+    total_earning = driver[:rides].sum do |ride|
+      ride[:cost]
+    end
+    {
+      id: driver[:id],
+      profits: total_earning,
+    }
+  end
+
+  result = driver_profits.max_by do |driver|
+    driver[:profits]
+  end
+  return "Driver ID: #{result[:id]}, has the highest earning of $#{result[:profits]}."
+
+  #   puts "#{driver_profits}"
+  #   return ""
+  #   most_profitable = nil
+  #   most_profits = 0
+  #   drivers.each do |driver|
+  #     total_earning = 0
+  #     driver[:rides].each do |ride_info|
+  #       total_earning += ride_info[:cost]
+  #     end
+
+  #     if total_earning > most_profits
+  #       most_profits = total_earning
+  #       most_profitable = driver[:id]
+  #     end
+  #   end
+  # return "Driver ID: #{most_profitable}, has earned the highest money of $#{most_profits}."
+end
+
+# 5 Which driver has the highest average rating?
+def highest_average_score(drivers)
+  driver_rating = drivers.map do |driver|
+    total_rating = driver[:rides].sum do |ride|
+      ride[:rate]
+    end
+    {
+      id: driver[:id],
+      average_rating: total_rating / driver[:rides].length,
```

Comment:
You should use floating point math here (see above for reasons).

The fix here is a little different though since you used `.sum` and not `.each`.  If you convert `total_rating` to a float before dividing you'll get back a float.
```suggestion
      average_rating: total_rating.to_f / driver[:rides].length,
```